### PR TITLE
feat: return 400 error code when not authorised endpoint /denylist/{type}/{id}

### PR DESCRIPTION
**Pull request by agusaldasoro**
This PR fixes https://github.com/decentraland/catalyst/issues/190

### CODE REVIEW

File: content/src/denylist/Denylist.ts [hunk 1]
```diff
@@ -116,3 +129,17 @@ export enum DenylistAction {
   ADDITION = 'addition',
   REMOVAL = 'removal'
 }
+
+export enum OperationStatus {
```
Comment (nchamo) [inline content/src/denylist/Denylist.ts:133]: Could we give more context on the name? Maybe something like `DenylistSignatureValidationStatus` and `DenylistSignatureValidationResult`?

File: content/src/denylist/Denylist.ts [hunk 2]
```diff
@@ -101,8 +113,9 @@ export class Denylist {
           this.network
         )
       })
+      return { status: OperationStatus.OK }
```
Comment (nchamo) [inline content/src/denylist/Denylist.ts:116]: I think we can avoid the try/catch here. Something like:
```
return new Promise((resolve) => {
validateSignature(
          metadata,
          messageToSign,
          resolve({ status: OperationStatus.OK }),
          errorMessage => resolve({ status: OperationStatus.ERROR, message: `Failed to authenticate the blocker. Error was: ${error}` }),
          (signer) => !!signer && (nodeOwner === signer || this.authenticator.isAddressOwnedByDecentraland(signer)),
          this.network
        )
})        
```

File: content/src/denylist/Denylist.ts [hunk 3]
```diff
@@ -26,9 +26,12 @@ export class Denylist {
     return `${actionMessage}-${target.asString()}-${timestamp}`
   }
 
-  async addTarget(target: DenylistTarget, metadata: DenylistMetadata) {
+  async addTarget(target: DenylistTarget, metadata: DenylistMetadata): Promise<OperationResult> {
     // Validate blocker and signature
-    await this.validateSignature(DenylistAction.ADDITION, target, metadata)
+    const operationResult: OperationResult = await this.validateSignature(DenylistAction.ADDITION, target, metadata)
+    if (isSuccessfulOperation(operationResult)) {
```
Comment (nchamo) [inline content/src/denylist/Denylist.ts:32]: What? If the signature validation is successful then you should continue and write everything to the database

File: content/src/denylist/Denylist.ts [hunk 4]
```diff
@@ -37,11 +40,15 @@ export class Denylist {
...
     })
+    return { status: OperationStatus.OK }
   }
 
-  async removeTarget(target: DenylistTarget, metadata: DenylistMetadata) {
+  async removeTarget(target: DenylistTarget, metadata: DenylistMetadata): Promise<OperationResult> {
     // Validate blocker and signature
-    await this.validateSignature(DenylistAction.REMOVAL, target, metadata)
+    const operationResult: OperationResult = await this.validateSignature(DenylistAction.REMOVAL, target, metadata)
+    if (isSuccessfulOperation(operationResult)) {
```
Comment (nchamo) [inline content/src/denylist/Denylist.ts:49]: Same as before

File: content/src/controller/Controller.ts [hunk 5]
```diff
@@ -540,14 +540,18 @@ export class Controller {
     }
 
     try {
-      await this.denylist.addTarget(target, { timestamp, authChain })
-      res.status(201).send()
+      const result: OperationResult = await this.denylist.addTarget(target, { timestamp, authChain })
+      if (isSuccessfulOperation(result)) {
+        res.status(201).send()
+      } else {
+        res.status(400).send(result.message)
```
Comment (nchamo) [inline content/src/controller/Controller.ts:547]: Didn't you have to change existing tests with this change? Didn't we have a test that verified when a denylist action failed?

File: content/src/denylist/Denylist.ts (hunk 2 above)
Comment (agusaldasoro) [reply content/src/denylist/Denylist.ts:116]: niceeee

File: content/src/controller/Controller.ts (hunk 5 above)
Comment (agusaldasoro) [reply content/src/controller/Controller.ts:547]: It wasn't an API test, it just checked that the service failed.

File: content/test/integration/denylist/Denylist.spec.ts [hunk 6]
```diff
@@ -100,6 +100,27 @@ describe('Integration - Denylist', () => {
...
+      contentPaths: ['content/test/integration/resources/some-binary-file.png']
+    })
+
+    // Deploy the entity
+    await server.deploy(deployData)
+
+    // Undenylist the entity
+    await server.undenylistEntity(entityBeingDeployed, decentralandIdentity)
+
+    // Assert that audit info marks the entity as denylisted
```
Comment (nchamo) [inline content/test/integration/denylist/Denylist.spec.ts:116]:
```suggestion
    // Assert that audit info marks the entity as not denylisted
```

**Review by nchamo — APPROVED** [review]
Great work!

**Pull request merged**